### ISSUE TITLE
Properly handle "unstructured data" from Calliope Mini

### ISSUE THREAD

**Issue by rzbrk** [calliope-edu/CalliopEO_AstroPi#48]
CalliopEO.py currently needs "line-oriented" data from the Calliope Mini. When Calliope Mini transmits "unstructured" data without newlines, CalliopEO.py will not continue processing the data (until it eventually receives a newline).

This could be improved by reading abitrary chunks of data from the serial input buffer. Later, CalliopEO.py can extract the lines and do further processing like adding the time stamp and so on.

Maybe, we can define this as a feature for a v2.

**Comment by rzbrk** [comment]
In my fork I created an [issue branch](https://github.com/rzbrk/CalliopEO_AstroPi/tree/issue48_unstr_data). There, I completely rewrote the function `readSerialData()` (see [here](https://github.com/rzbrk/CalliopEO_AstroPi/blob/cc6808d0ece5f7d0eb5e2b1205b164fa5c657beb/CalliopEO.py#L107)) to handle "unstructured data", in particular a stream of data not containing newlines or very, very long lines.

I defined a serial data buffer (see [here](https://github.com/rzbrk/CalliopEO_AstroPi/blob/cc6808d0ece5f7d0eb5e2b1205b164fa5c657beb/CalliopEO.py#L110)) and new global variable `MAX_LINE_LENGTH` (see [here](https://github.com/rzbrk/CalliopEO_AstroPi/blob/cc6808d0ece5f7d0eb5e2b1205b164fa5c657beb/CalliopEO.py#L57)). If the buffer length exceeds `MAX_LINE_LENGTH` the buffer is "cropped" by extracting `MAX_LINE_LENGTH` characters as a line which will be outputted to the file.

The functions `waitSerialStart()` and `waitSerialEnd()` are now incorporated in `readSerialData()`.

The return value of this function is no longer a list of lines. The data is directly written to file (via function `write2File()`). The return value of `readSerialData()` is now the number of characters written to the file.

Before merging this change to main, this needs further testing.
